fc-community-backend/apps/shared/src/web3.ts

```diff
@@ -80,31 +80,20 @@ export class FactChainBackend implements NoteReader, NoteWriter {
     const currentBlockNumber = await this._provider.getBlockNumber();
     const today = new Date();
     const from = new Date(today.getTime() - lookBackDays * 24 * 60 * 60 * 1000);
-    console.log(`getting notes between ${from} and ${today}`);
-
-    const block_periods = timePeriodToBlockPeriods(
-      from,
-      today,
-      currentBlockNumber,
-    );
+    const blockPeriods = timePeriodToBlockPeriods(from, today, currentBlockNumber);
 
-    let notePromises: Promise<Note>[] = [];
-    for (const period of block_periods) {
+    console.log(`getting notes between ${from} and ${today}`);
+    const notePromises = blockPeriods.flatMap(async (period) => {
       const events = await this.getEvents("NoteCreated", period[0], period[1]);
-      const relatedEvents = events.filter((e) =>
-        predicate(e.args[0], e.args[1]),
+      const relatedEvents = events.filter((e) => predicate(e.args[0], e.args[1]));
+      return Promise.all(
+        relatedEvents.map(async (event) => {
+          return await this.getNote(event.args[0], event.args[1]);
+        })
```

Comment:
pretty sure you can simplify to 
```typescript
      return Promise.all(relatedEvents.map((event) => this.getNote(event.args[0], event.args[1]))
```

Reply:
agreed, `this.getNote` is already an async function 👍